finetrainers/hunyuan_video/hunyuan_video_lora.py

```diff
@@ -35,24 +34,41 @@ def load_components(
     text_encoder_2 = CLIPTextModel.from_pretrained(
         model_id, subfolder="text_encoder_2", torch_dtype=text_encoder_2_dtype, revision=revision, cache_dir=cache_dir
     )
-    transformer = HunyuanVideoTransformer3DModel.from_pretrained(
-        model_id, subfolder="transformer", torch_dtype=transformer_dtype, revision=revision, cache_dir=cache_dir
-    )
-    vae = AutoencoderKLHunyuanVideo.from_pretrained(
-        model_id, subfolder="vae", torch_dtype=vae_dtype, revision=revision, cache_dir=cache_dir
-    )
-    scheduler = FlowMatchEulerDiscreteScheduler()
     return {
         "tokenizer": tokenizer,
         "text_encoder": text_encoder,
         "tokenizer_2": tokenizer_2,
         "text_encoder_2": text_encoder_2,
-        "transformer": transformer,
-        "vae": vae,
-        "scheduler": scheduler,
     }
 
 
+def load_latent_models(
+    model_id: str = "tencent/HunyuanVideo",
+    vae_dtype: torch.dtype = torch.float16,
+    revision: Optional[str] = None,
+    cache_dir: Optional[str] = None,
+    **kwargs,
+) -> Dict[str, nn.Module]:
+    vae = AutoencoderKLHunyuanVideo.from_pretrained(
+        model_id, subfolder="vae", torch_dtype=vae_dtype, revision=revision, cache_dir=cache_dir
+    )
+    return {"vae": vae}
+
+
+def load_diffusion_models(
+    model_id: str = "tencent/HunyuanVideo",
+    transformer_dtype: torch.dtype = torch.bfloat16,
+    revision: Optional[str] = None,
+    cache_dir: Optional[str] = None,
+    **kwargs,
+) -> Dict[str, nn.Module]:
+    transformer = HunyuanVideoTransformer3DModel.from_pretrained(
+        model_id, subfolder="transformer", torch_dtype=transformer_dtype, revision=revision, cache_dir=cache_dir
+    )
+    scheduler = FlowMatchEulerDiscreteScheduler()
+    return {"transformer": transformer, "scheduler": scheduler}
+
+
 def initialize_pipeline(
```

Comment:
Does not really require any changes. All the currently loaded models will be passed in (if no precomputation, then all text encoders, vae and transformer. if precomputation, then only transformer). The remaining models that are not provided will be loaded with defaults because they are all needed for inference.

One thing I still have in mind is implementing precomputation for validation prompts, but that's something I will revisit in a future PR because I'm busy with a few other things on diffusers-side that I would like to work on first

Reply:
Yeah I guess that is an extreme case. Folks that can afford to do validation during training, would probably have the GPU resources to run without precomputing the conditions for validation. If it becomes a problem, we can revisit but for now, without it is enough IMO.